AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/handler/ResHandler.java

```diff
@@ -100,4 +110,52 @@ public void process(Element element, EComponentHolder holder) {
 	protected boolean hasContextCompatInClasspath() {
 		return processingEnvironment().getElementUtils().getTypeElement(CanonicalNameConstants.CONTEXT_COMPAT) != null;
 	}
+	
+	protected boolean shouldUseContextGetDrawableMethod() {
+		return androidManifest.getMinSdkVersion() >= MIN_SDK_WITH_CONTEXT_GET_DRAWABLE;
+	}
+
+	protected boolean hasGetDrawableInContext() {
+		TypeElement context = processingEnvironment().getElementUtils().getTypeElement(CanonicalNameConstants.CONTEXT);
+
+		return hasGetDrawable(context);
+	}
+
+	private void handleDrawableCompatibility(EComponentHolder holder, JFieldRef ref, JBlock methodBody, JFieldRef idRef, String resourceMethodName) {
+		if (hasContextCompatInClasspath()) {
+			methodBody.assign(ref, classes().CONTEXT_COMPAT.staticInvoke("getDrawable").arg(holder.getContextRef()).arg(idRef));
+		} else if (shouldUseContextGetDrawableMethod() && !hasContextCompatInClasspath()) {
+			methodBody.assign(ref, holder.getContextRef().invoke("getDrawable").arg(idRef));
+		} else if (!shouldUseContextGetDrawableMethod() && hasGetDrawableInContext() && !hasContextCompatInClasspath()) {
+			createCallWithIfGuard(holder, ref, methodBody, idRef);
+		} else {
+			methodBody.assign(ref, invoke(holder.getResourcesRef(), resourceMethodName).arg(idRef));
+		}
+	}
+
+	private void createCallWithIfGuard(EComponentHolder holder, JFieldRef ref, JBlock methodBody, JFieldRef idRef) {
+		JVar resourcesRef = holder.getResourcesRef();
+		JConditional guardIf = methodBody._if(holder.classes().BUILD_VERSION.staticRef("SDK_INT").gte(holder.classes().BUILD_VERSION_CODES.staticRef("LOLLIPOP")));
+		JBlock ifBlock = guardIf._then();
+		ifBlock.assign(ref, holder.getContextRef().invoke("getDrawable").arg(idRef));
+
+		JBlock elseBlock = guardIf._else();
+		elseBlock.assign(ref, resourcesRef.invoke("getDrawable").arg(idRef));
+	}
+
+	private boolean hasGetDrawable(TypeElement type) {
+		if (type == null) {
+			return false;
+		}
+
+		List<? extends Element> allMembers = processingEnvironment().getElementUtils().getAllMembers(type);
+		for (Element element : ElementFilter.methodsIn(allMembers)) {
+			ExecutableElement executableElement = (ExecutableElement) element;
```

Comment:
This is not needed since, `ElementFilter.methodsIn` already returns `ExecutableElement`s.